src/components/remote_control/src/commands/base_command_request.cc

```diff
@@ -0,0 +1,583 @@
+      SendResponse(false, result_codes::kRejected, "");
+      return false;
+    }
+  }
+
+  return false;
+}
+
+bool BaseCommandRequest::IsAutoAllowed(
+    application_manager::TypeAccess access) const {
+  return access == application_manager::kAllowed;
+}
+
+void BaseCommandRequest::SendDisallowed(
+    application_manager::TypeAccess access) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  std::string info;
+  switch (access) {
+    case application_manager::kAllowed:
+      return;
+    case application_manager::kDisallowed:
+      info = disallowed_info_.empty()
+                 ? "The RPC is disallowed by vehicle settings"
+                 : disallowed_info_;
+      break;
+    default:
+      info = "Unknown issue";
+  }
+  LOG4CXX_ERROR(logger_, info);
+  SendResponse(false, result_codes::kDisallowed, info);
+}
+
+void BaseCommandRequest::SendGetUserConsent(const Json::Value& value) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  DCHECK(app_);
+  Json::Value params;
+  params[json_keys::kAppId] = app_->hmi_app_id();
+  params[message_params::kModuleType] = ModuleType(value);
+  SendRequest(functional_modules::hmi_api::get_user_consent, params);
+}
+
+std::string BaseCommandRequest::ModuleType(const Json::Value& message) {
+  return "";
+}
+
+std::vector<std::string> BaseCommandRequest::ControlData(
+    const Json::Value& message) {
+  return std::vector<std::string>();
+}
+
+void BaseCommandRequest::on_event(
+    const rc_event_engine::Event<application_manager::MessagePtr, std::string>&
+        event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  SetResourceState(msg_json_, ResourceState::FREE);
+
+  if (event.id() == functional_modules::hmi_api::get_user_consent) {
+    ProcessAccessResponse(event);
+  } else {
+    OnEvent(event);  // run child's logic
+  }
+}
+
+void BaseCommandRequest::ProcessAccessResponse(
+    const rc_event_engine::Event<application_manager::MessagePtr, std::string>&
+        event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (!app_) {
+    LOG4CXX_ERROR(logger_, "Application doesn't registered!");
+    SendResponse(false, result_codes::kApplicationNotRegistered, "");
+    return;
+  }
+
+  application_manager::Message& hmi_response = *(event.event_message());
+  const application_manager::MessageValidationResult validate_result =
+      service_->ValidateMessageBySchema(hmi_response);
+  LOG4CXX_DEBUG(logger_,
+                "HMI response validation result is " << validate_result);
+
+  if (validate_result !=
+      application_manager::MessageValidationResult::SUCCESS) {
+    SendResponse(
+        false, result_codes::kGenericError, "HMI has sent invalid parameters");
+    return;
+  }
+
+  Json::Value value;
+  Json::Reader reader;
+  reader.parse(event.event_message()->json_message(), value);
+
+  std::string result_code;
+  std::string info;
+  const bool is_succeeded = ParseResultCode(value, result_code, info);
+
+  bool is_allowed = false;
+  if (is_succeeded) {
+    if (IsMember(value[kResult], message_params::kAllowed) &&
```

Comment:
kAllowed is used in 3 different scopes in this file? used as message_params::kAllowed, application_manager::TypeAccess::kAllowed, and application_manager::kAllowed.

Reply:
@JackLivio RC have to access to application_manager::TypeAccess::kAllowed, and application_manager::kAllowed.